### ISSUE TITLE
Java websocket server sending the ping message but go lang webscket client not received the ping message

### ISSUE THREAD

**Issue by vitpravin** [eclipse-ee4j/websocket-api#367]
From java  server-side sending the ping  message but the go lang client is not able to receive the ping message 
Go lang client destails - https://github.com/gorilla/websocket

Same go lang websocket client working the node js server for a ping message
Same java websocket server (using this lib only) working with node js client for a ping message 

The only issue we are facing is for the **java WebSocket server and go, lang websocekt client**

**Java server websocket code to send the ping message to the client** 

Session session = client session object that we need to send the ping message
session.getBasicRemote().sendPing(ByteBuffer.wrap("clientid".getBytes()));

Basically, I want to understand why this ping message issue with java and go lang only
I have tested with java and go lang code with different client and server 
ping functionality working as expected without any code changes in java and go lang 
as of now only suspecting an issue with java and go lang

could you please help with this because this is a specific issue I tried multiple things but was not able to figure out the issue?

Thanks

**Comment by markt-asf** [comment]
Again, we can't help you. This project has not provided the implementation you are using. You need to contact the provider of your Java WebSocket implementation which is NOT this project.

**Comment by vitpravin** [comment]
Thanks for the quick reply @markt-asf but I am not understanding what you want to say

I am using this project implementation provided by this lib only 

code details in a project to send the ping message
https://github.com/eclipse-ee4j/websocket-api/blob/master/api/client/src/main/java/jakarta/websocket/RemoteEndpoint.java#L108

  /**
     * Send a Ping message containing the given application data to the remote endpoint. The corresponding Pong message
     * may be picked up using the MessageHandler.Pong handler.
     * <p>
     * It is not safe for other threads to use the ByteBuffer until the sending of this message is complete. If the
     * sending of the message completes successfully, the buffer's limit will be unchanged and the buffer's position
     * will be equal to the limit. If the sending of the message does not complete successfully, the state of the buffer
     * is undefined.
     *
     * @param applicationData the data to be carried in the ping request.
     * @throws IOException              if the ping failed to be sent
     * @throws IllegalArgumentException if the applicationData exceeds the maximum allowed payload of 125 bytes
     */
    void sendPing(ByteBuffer applicationData) throws IOException, IllegalArgumentException;

**Comment by markt-asf** [comment]
Look at the code you linked. It only defines the interface. There is no implementation. The implementation will be provided by a different project. You need to raise your issue with the project that provides the implementation you are using.

If you are running in a container (Tomcat, Jetty, Glassfish, etc.) the container will be providing the implementation. If you are running standalone you you must know which JAR you have added that contains the WebSocket implementation.

**Comment by joakime** [comment]
@vitpravin I have to agree with @markt-asf 

This is an inappropriate place to file an issue with an implementation of this spec/api.
If you look at the code in this repository, you'll see that its just discussing HOW an implementation would implement the spec/api that is contained within this repository.

Also note that there are standalone implementations of this spec/api, such as Tyrus (https://github.com/eclipse-ee4j/tyrus)
